lib/plausible/stats/timeseries.ex

```diff
@@ -24,8 +24,9 @@ defmodule Plausible.Stats.Timeseries do
 
     Enum.map(steps, fn step ->
       empty_row(step, metrics)
-      |> Map.merge(Enum.find(event_result, fn row -> row[:date] == step end) || %{})
-      |> Map.merge(Enum.find(session_result, fn row -> row[:date] == step end) || %{})
+      |> Map.merge(Enum.find(event_result, fn row -> is_equal.(row[:date], step) end) || %{})
```

Comment:
Instead of each `buckets/1` variant returning its own `is_equal` and `format` functions, how about adding some generic versions of both:

```elixir
defp date_eq(%DateTime{} = left, %DateTime = right) do
  NaiveDateTime.compare(left, right) == :eq
end

defp date_eq(%Date{} = left, %Date = right) do
  Date.compare(left, right) == :eq
end

defp date_eq(left, right) do
  left == right
end

defp date_format(%DateTime{} = date) do
  Timex.format!(date, "{YYYY}-{0M}-{0D} {h24}:{m}:{s}")
end

defp date_format(date) do
  date
end
```

Reply:
done in https://github.com/ruslandoga/analytics/pull/75/commits/3467d6261dbbb61f8281ffe06c4f78c23a267273